BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h

```diff
@@ -31,20 +31,28 @@
 - (void)authenticateInAppWithCompletionBlock:(void (^)(BOXUser *user, NSError *error))completion;
 
 /**
- *  Discerns whether a launch URL is associated with Box authentication - if it is the return URL that should be used
+ *  Discerns whether a launch URL is associated with Box app to app authentication - if it is the return URL that should be used
  *  to complete the user's authentication in the case of authenticating App-to-App using the Box app.
  *
  *  @param authenticationURL    The URL with which the app was launched.
  */
-+ (BOOL)canCompleteAuthenticationWithURL:(NSURL *)authenticationURL;
++ (BOOL)canCompleteAppToAppAuthenticationWithURL:(NSURL *)authenticationURL;
 
 /**
- *  Complete the user's authentication to the account that is in use in the Box app.
+ *  Complete the user's authentication to the account that is in use in the Box app.(App-to-App)
  *  On completion, the original completionBlock will be called.
  *
  *  @param authenticationURL    The URL with which the app was launched.
  */
-+ (void)completeAuthenticationWithURL:(NSURL *)authenticationURL;
++ (void)completeAppToAppAuthenticationWithURL:(NSURL *)authenticationURL;
+
+/**
+ *  Complete the user's authentication from the URL with which the app was launched.
+ *
+ *  @param authenticationURL    The URL with which the app was launched.
+ *  @param completionBlock      Called when the authentication has completed.
+ */
+- (void)completeAuthenticationWithURL:(NSURL *)authenticationURL completionBlock:(void (^)(BOXUser *user, NSError *error))completion;
```

Comment:
We discussed offline to not expose this publicly. I have moved it to a private header.